### PR TITLE
Types stripe tax rates

### PR DESCRIPTION
**Pull request by Stilborg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://stripe.com/docs/api/tax_rates>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -37,87 +37,93 @@
 /// <reference types="node" />
 
 declare class Stripe {
-    DEFAULT_HOST: string;
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: This file uses 4 space indents - can you avoid making unrelated changes here?

Comment (Stilborg) [reply types/stripe/index.d.ts]: Ups, sorry about that

File: types/stripe/index.d.ts [hunk 2]
```diff
@@ -6734,6 +6750,108 @@ declare namespace Stripe {
         }
     }
 
+    namespace taxRates {
+        interface ITaxRateCreationOptions {
+          /**
+           * The display name of the tax rate, which will be shown to users.
+           */
+            display_name: string;
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: nit: the indentation for these values comments doesn't align with the comments. I think you have an extra space everywhere.

File: types/stripe/index.d.ts [hunk 3]
```diff
@@ -6734,6 +6750,108 @@ declare namespace Stripe {
...
+          /**
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: nit: add newlines in between properties.

File: types/stripe/index.d.ts [hunk 4]
```diff
@@ -6734,6 +6750,108 @@ declare namespace Stripe {
...
+           * The jurisdiction for the tax rate.
+           */
+          jurisdiction?: string;
+          /**
+           * Set of key-value pairs that you can attach to an object. This can be useful for storing additional information about the object in a structured format.
+           * Individual keys can be unset by posting an empty value to them. All keys can be unset by posting an empty value to metadata.
+           */
+          metadata?: IMetadata;
+        }
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: nit: add newlines in between interfaces

File: types/stripe/index.d.ts [hunk 5]
```diff
@@ -10817,6 +10935,53 @@ declare namespace Stripe {
             retrieve(id: string, data?: sources.ISourceRetrieveOptions, response?: IResponseFn<sources.ISource>): Promise<sources.ISource>;
         }
 
+        class TaxRates extends StripeResource {
+          /** Create TaxRate: https://stripe.com/docs/api/tax_rates/create */
+          create(
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: This is fine, since these are overloads of the same method.

File: types/stripe/stripe-tests.ts [hunk 6]
```diff
@@ -1989,3 +1989,26 @@ stripe.charges.create({
...
+    description: 'VAT Germany',
+    jurisdiction: 'DE',
+    percentage: 19.0,
+    inclusive: false,
+}).then(
+    tr => {stripe.taxRates.update(tr.id, { metadata: { order_id: '6735' } })
+        .then(tru => { stripe.taxRates.retrieve(tru.id)
+            .then(trr => {
+                stripe.taxRates.list({ limit: 3 })
+                .then(trl => {
```
Comment (vaskevich) [inline types/stripe/stripe-tests.ts]: Can you use `// $ExpectType` here?

**Review by vaskevich — APPROVED** [review]
LGTM mod nits

**Comment by DanielRosenwasser** [comment]
Mind addressing the nits? If you address that I can pull in the changes.

File: types/stripe/index.d.ts [hunk 7]
```diff
@@ -9845,48 +10862,145 @@ declare namespace Stripe {
...
+                data: customerTaxIds.ITaxIdCreationOptions,
+                response?: IResponseFn<subscriptions.ISubscription>,
+            ): Promise<customerTaxIds.ITaxId>;
+
+            listTaxIds(
+                customerId: string,
+                data: IListOptions,
+                options: HeaderOptions,
+                response?: IResponseFn<IList<sources.ISource>>,
+            ): IListPromise<sources.ISource>;
```
Comment (Jank1310) [inline types/stripe/index.d.ts]: Are the return types for listTaxIds correct? Maybe better return ```customerTaxIds.ITaxId```?

Comment (Stilborg) [reply types/stripe/index.d.ts]: I think you might be right @Jank1310 , but it is not related to the topic of the PR. 
I added support for tax rates, customer tax id's are a different thing. 
But I ran prettier so it looks like I changed those lines as well, sorry for that.

File: types/stripe/index.d.ts [hunk 8]
```diff
@@ -819,7 +827,7 @@ declare namespace Stripe {
              * The individual’s gender (International regulations require either “male” or “female”).
              * This can be unset by updating the value to null and then saving.
              */
-            gender?: "male" | "female";
```
Comment (vaskevich) [inline types/stripe/index.d.ts]: Can you keep using double quotes to avoid a giant diff here?

Comment (vaskevich) [reply types/stripe/index.d.ts]: Please keep the PR diff limited to places that you're actually materially changing.

Comment (Stilborg) [reply types/stripe/index.d.ts]: @vaskevich I ran prettier using the prettierrc.json in the root. That config calls for singleQuotes. 
Doing changes to this file has proven quite painful without the assistance of a formatter.
Is there an overall plan for formatting on DT?

For this PR, I will start over with double quotes and no prettier when I find the time

Comment (vaskevich) [reply types/stripe/index.d.ts]: Maybe you can send out a PR with just the prettier changes that can be auto-approved? And then follow up with this change after.

Comment (Stilborg) [reply types/stripe/index.d.ts]: Good point @vaskevich thank you, I will do that.

**Comment by Stilborg** [comment]
@vaskevich ok, this one is back :)
